### ISSUE TITLE
False positive error from JavadocType with record

### ISSUE THREAD

**Issue by turbanoff** [checkstyle/checkstyle#13309]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadoctype.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac YOUR_FILE.java

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="allowMissingParamTags" value="true"/>
            <property name="allowUnknownTags" value="true"/>
        </module>
    </module>
</module>

/var/tmp $ cat MyTransaction.java
package bkk;

/**
 * One Transaction.
 *
 * @param transactionId  unique ID of the transaction in format: {@code <first Code>:<second Code>}
 */
public record MyTransaction(String transactionId) {
}

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-10.12.1-all.jar -c config.xml MyTransaction.java
Starting audit...
[ERROR] D:\Projects\app4Test\checkstyle-test\src\main\java\bkk\MyTransaction.java:6:4: Unused @param tag for 'transactionId'. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.
```
---

I expect no warning or errors here. Javadoc is properly written in the proper place.

---

**Comment by turbanoff** [comment]
Here
![image](https://github.com/checkstyle/checkstyle/assets/741251/b01d6e21-0e74-47a3-a295-498d0e2bca13)
[MyTransaction.zip](https://github.com/checkstyle/checkstyle/files/11870164/MyTransaction.zip)

**Comment by romani** [comment]
I am on phone, please show html on constructor detailed section.

**Comment by turbanoff** [comment]
Constructor + getter.
![изображение](https://github.com/checkstyle/checkstyle/assets/741251/9544eaf5-41ea-4077-9c9b-8fb3b1d48123)

**Comment by sktpy** [comment]
**Preliminary Investigation:**
While playing around with config and input, I have discovered that this isn't the issue for every record, I have narrowed it down to when the description of `@param` tag contains atleast one pair of angular bracket enclosing anything (but not empty) e.g. `<stuff>`

**config.xml**
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocType"/>
  </module>
</module>
```

**Amongus.java**
```JAVA
/**
 *
 * @param sus <probably red>
 */
record CaughtVenting(String sus) {
}

/**
 *
 * @param safe will probably be ghost next round
 */
record DidTheMedBayScan(String safe) {
}

/**
 *
 * @param idk <>
 */
record AfkInTheLobby(String idk) {
}
```

**CLI:**
```
java -jar checkstyle-10.14.0-all.jar -c test/config.xml test/Amongus.java
[ERROR] Amongus.java:3:4: Unused @param tag for 'sus'. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

We only got false positive for the `@param` tag with description having the said pattern (i.e. pair of angular brackets with stuff inside), rest two case without angular brackets pair, empty angular brackets pair are fine.

Just putting it out here, will dive deeper once I start sussing out the impostor among us.

**Comment by sktpy** [comment]
I am on it ඞ

**Comment by sktpy** [comment]
The fix is done, but waiting for the resolution of the issue above to decide whether should I send the PR with one where this false negative is handled(much cleaner code) or the one where it should be preserved(only requires minor change but still leaves us with the complex code(in comparison to the alternative) which led to the bug in first place.

**Comment by nrmancuso** [comment]
@sktpy 

>The fix is done, but waiting for the resolution of the issue above to decide whether should I send the PR:

If you have the "cleaner" code already completed, please link by commit to https://github.com/checkstyle/checkstyle/issues/14581

I would personally go with 

>In which that false negative is preserved which only needed a minor change but still leaves us with the complex code(in comparison to the alternative) which led to the bug in first place.

To get this issue closed quickly. #14581 may not be approved for some time. Either way, incremental improvements are the best way to quickly make progress, and it would be good to get this fix to users sooner.

**Comment by sktpy** [comment]
> incremental improvements are the best way to quickly make progress

:star: 
<hr>

**1.**  Alternative cleaner fix: **https://github.com/sktpy/checkstyle/pull/1**
Have created a PR for this in my own fork, to not have it confused with the other PR, while also allowing us to have a discussion regarding that in there.

**2.** As recommended, fix which preserves the false negative of the https://github.com/checkstyle/checkstyle/issues/14581, have created a PR for this here: **https://github.com/checkstyle/checkstyle/pull/14603**

After report generation, and adding explanatory comment will move them out of draft mode.

**Comment by sktpy** [comment]
We should go with our alternative cleaner fix due to the following reasons:
1. It is cleaner.
2. It does not have anything to do with generation of the new violation as explained in https://github.com/sktpy/checkstyle/pull/1#discussion_r1511710537
3. It does exactly what it is supposed to, i.e. to extract param name, nothing less, nothing more.

We should get rid of the buggy implementation all together as done in alternative PR,
and if we want to not have that violation, we should be investigating what is the actual cause of it in a PR that resolves the referenced issue https://github.com/checkstyle/checkstyle/issues/14581,
because the change of logic in the method isn't the cause, it just highlighted the violation which was able to sweep under the radar previously.

**Comment by nrmancuso** [comment]
Closed via #14603
